librz/asm/arch/hexagon/hexagon.h

```diff
@@ -46,237 +51,521 @@ typedef enum {
 	HEX_OP_REG_HI = 1 << 1, // Rn.H marker
 	HEX_OP_REG_LO = 1 << 2, // Rn.L marker
 	HEX_OP_REG_PAIR = 1 << 3, // Is this a register pair?
+	HEX_OP_IMM_SCALED = 1 << 4 // Is the immediate shifted?
 } HexOpAttr;
 
+typedef enum {
+	HEX_NO_LOOP = 0,
+	HEX_LOOP_0 = 1, // Is packet of loop0
+	HEX_LOOP_1 = 1 << 1, // Is packet of loop1
+	HEX_ENDS_LOOP_0 = 1 << 2, // Packet ends loop0?
+	HEX_ENDS_LOOP_1 = 1 << 3, // Packet ends loop1?
+} HexLoopAttr;
+
+typedef struct {
+	bool first_insn;
+	bool last_insn;
+	bool valid_pkt;
+	ut32 pkt_addr;
+	char syntax_prefix[8]; // Package indicator
+	char syntax_postfix[16]; // for ":endloop" string.
+	unsigned int parse_bits;
+	HexLoopAttr loop_attr;
+} HexPktInfo;
+
 typedef struct {
 	ut8 type;
 	union {
 		ut8 reg; // + additional Hi or Lo selector // + additional shift // + additional :brev //
-		ut32 imm;
-		ut8 pred; // predicates - P0-P3 registers
-		ut8 cr; // control register
-		ut8 sys; // system control register
+		st64 imm;
 	} op;
 	ut8 attr;
+	ut8 shift;
 } HexOp;
 
 typedef struct {
 	int instruction;
 	ut32 mask;
-	HexPred predicate; // predicate set if set
```

Comment:
Why are predicates no longer in use? They are pretty common in the code.

Reply:
Oh, I quite frankly forgot about them. Will implement as soon as possible.

Reply:
Fixed with https://github.com/rizinorg/rizin/pull/1614/commits/1435be081c4e3265aa820706993112eb78670499

Reply:
Although https://github.com/rizinorg/rz-hexagon/issues/29 still persists.